intro/README.md

```diff
@@ -0,0 +1,561 @@
+With no further ado, let us take a random C program and see what it takes to compile it with the D compiler:
+
+```c
+#include <stdio.h>
+
+int main()
+{
+    int position = 7, c, n = 10;
+    int array[n] = {1, 2, 3, 4, 5, 6, 7, 8, 9, 10};
+
+    for (c = position - 1; c < n - 1; c++)
+        array[c] = array[c+1];
+
+    printf("Resultant array:\n");
+    for (c = 0; c < n - 1; c++)
+        printf("%d\n", array[c]);
+
+    return 0;
+}
+```
+
+The code above simply deletes an element in an array.
+Now let's take a look on what the minimum modifications are to make the code compile and run with D:
+
+```d
+import std.stdio;
+
+int main()
+{
+    int position = 7, c, n = 10;
+    int[10] array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+    for (c = position - 1; c < n - 1; c++)
+        array[c] = array[c+1];
+
+    printf("Resultant array:\n");
+    for (c = 0; c < n - 1; c++)
+        printf("%d\n", array[c]);
+
+    return 0;
+}
+```
+    
+As you can see, the only differences are:
+
+- the `#include` directive was replaced by an `import` statement;
+- the array definition and initialization are slightly modified;
+
+Most C programs require minimal changes in order to be compiled with the D compiler.
+So do not worry, even if you don't have any previous experience in D, you will be able to understand most of the programs written in it because the syntax is extremely similar to the C one.
+
+Now, using the above code snippet, let us delve into the D specific concepts.
+
+## Imports
+
+In D, **imports** represent the counterpart of the C **include** directive
+However the are 2 fundamental differences:
+
+1. Imports may selectively specify which symbols are to be imported. For example, in the above code snippet, the full standard IO module of the standard library is imported, even though only the **printf** function is used.
+This results in a degradation in compile time since there is a larger symbol pool that needs to be examined when trying to resolve a symbol.
+In order to fix this, we can replace the blunt import with:
+```d
+import std.stdio : printf;
+```
+
+1. Imports may be used at any scope level.
+```d
+int main()
+{
+    int position = 7, c, n = 10;
+    int[10] array = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+
+    for (c = position - 1; c < n - 1; c++)
+        array[c] = array[c+1];
+
+    // If the lines calling printf are deleted,
+    // it is easier to spot the now useless import
+    import std.stdio : printf;
+    printf("Resultant array:\n");
+    for (c = 0; c < n - 1; c++)
+        printf("%d\n", array[c]);
+
+    return 0;
+}
+```
+
+### C to D
+
+Go to this [link](https://github.com/TheAlgorithms/C/tree/master/searching). You will find a series of searching algorithms, each implemented in C in its own file.
+- Choose one algorithm and port it to D with minimum modifications.
+- Update the code using D specific features to improve the code (fewer lines of code, increase in expressiveness etc.).
```

Comment:
This should be done later. Right now we haven't shown anything about how powerful D is.